ui/tests/integration/components/sync/secrets/page/destinations-test.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupEngine } from 'ember-engines/test-support';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { render, click } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { allowAllCapabilitiesStub } from 'vault/tests/helpers/stubs';
+import sinon from 'sinon';
+
+module('Integration | Component | sync | Page::Destinations', function (hooks) {
+  setupRenderingTest(hooks);
+  setupEngine(hooks, 'sync');
+  setupMirage(hooks);
+
+  hooks.beforeEach(function () {
+    this.owner.lookup('service:version').version = '1.16.0+ent';
+
+    this.server.post('/sys/capabilities-self', allowAllCapabilitiesStub());
+
+    const store = this.owner.lookup('service:store');
+    const modelName = 'sync/destinations/aws-sm';
+    store.pushPayload(modelName, {
+      modelName,
+      ...this.server.create('sync-destination', 'aws-sm'),
+    });
+
+    this.destinations = store.peekAll(modelName).toArray();
+    this.destinations.meta = {
+      filteredTotal: this.destinations.length,
+      currentPage: 1,
+      pageSize: 5,
+    };
+
+    this.renderComponent = () => {
+      return render(
+        hbs`<Secrets::Page::Destinations
+          @destinations={{this.destinations}}
+          @typeFilter={{this.typeFilter}}
+          @nameFilter={{this.nameFilter}}
+        />`,
+        { owner: this.engine }
+      );
+    };
+
+    this.transitionStub = sinon.stub(this.owner.lookup('service:router'), 'transitionTo');
+    this.clearDatasetStub = sinon.stub(store, 'clearDataset');
+  });
+
+  test('it should render header and tabs', async function (assert) {
+    await this.renderComponent();
+    assert.dom('[data-test-crumb="0"]').includesText('Secrets sync', 'Breadcrumb renders');
+    assert.dom('[data-test-page-title]').hasText('Secrets sync', 'Page title renders');
+    assert.dom('[data-test-tab="Overview"]').exists('Overview tab renders');
+    assert.dom('[data-test-tab="Destinations"]').exists('Destinations tab renders');
```

Comment:
Some of these selectors are already available in the test helper, feel free to add these there :) 

Selector variables have made test updates so much easier when updating 50+ files during HDS changes, so I think it's a good pattern to maintain!

Reply:
HDS work is just an example of updating many selectors in one go. But this could happen anytime major code refactors happen. I've spent lots of time updating miscellaneous selectors across files, and appreciated when all I need to do is update one selector. IMO expediting this process is worth a quick reference to another file to find the selector in the DOM.

But I agree, digging through lots of files is a pain which is why I think it's important to keep selectors in just **one** file streamlines the process. PKI is a mess with each component having its own selector file. 

I'm not committed to the pattern, it's just made my life easier which is why I mentioned it. Probably worth bringing up with the rest of the team to see how folks feel just so we're consistent. Happy to defer to whatever the majority preference is here!

Reply:
Ok that all makes sense. Since you have established the pattern in sync I will update the tests I've made (in another PR) to follow it and then follow up with the team and discuss ways of standarizing things since everyone does it a bit different.

My personal feeling is that this pattern feels a lot like using `ember-cli-page-object` which I've never been a fan of but maybe we can come up with a system that fits somewhere in the middle.